[new product] apache flink (#5155)
captn3m0/endoflife.date: products/apache-flink.md

---
title: Apache Flink
category: server-app

> tags: apache streaming batch-processing

Reviewer: We're conservative about adding new tags. As per endoflife.date/contribute:

> Remember that no tag is better than a useless tag. So do not introduce new tags when adding a product
> and use one of the tags listed on https://endoflife.date/tags/.
>
> Should you want to add a new tag, please open an issue first to discuss it with the team.
> Moreover, any new tag must be applied in a single PR to all products that should have it.

```suggestion
tags: apache java-runtime
```

Switching to the java-runtime, since flink runs on the JVM.

Author: ack, appreciated

permalink: /apache-flink
alternate_urls:
- /flink
releasePolicyLink: https://hub.docker.com/_/flink # This is the most conclusive resource
changelogTemplate: https://nightlies.apache.org/flink/flink-docs-master/release-notes/flink-+__LATEST__
releaseDateColumn: true

auto:
  methods:
  -   git: https://github.com/apache/flink.git

releases:
-   releaseCycle: "1.19"
    releaseDate: 2024-03-15
    eol: false
    latest: "1.19.0"
    latestReleaseDate: 2024-03-21

> -   releaseCycle: "1.8"
>     releaseDate: 2023-10-24
>     eol: false
>     latest: "1.8.1"
>     latestReleaseDate: 2024-03-21

Reviewer: 1.18, not 1.8

```suggestion
-   releaseCycle: "1.18"
    releaseDate: 2023-10-24
    eol: false
    latest: "1.18.1"
    latestReleaseDate: 2024-01-16
```

> -   releaseCycle: "1.7"
>     releaseDate: 2023-04-28
>     eol: false
>     latest: "1.7.2"
>     latestReleaseDate: 2023-03-23

Reviewer: 1.17, not 1.7

```suggestion
-   releaseCycle: "1.17"
    releaseDate: 2023-03-23
    eol: false
    latest: "1.17.2"
    latestReleaseDate: 2023-11-27
```

---

> [Apache Flink](https://flink.apache.org/) is a  is a stream processing
> framework designed for processing large-scale data streams in real-time with
> high throughput and low latency.

> Apache Flink maintains match releases for multiple versions. The most conclusive
> documentation for supported versions is referenced in their official dockerhub
> [image documentation](https://hub.docker.com/_/flink).

Reviewer: The Docker Hub link is anyway shown via `releasePolicyLink`.

```suggestion
Apache Flink maintains match releases for multiple versions. Backports seem to be
maintained for the latest 3 releases.

## [Java Compatibility](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/deployment/java_compatibility/)

- Java 8 is supported, but deprecated.
- Java 11 is the recommended version.
- Java 17 is supported experimentally for Flink 1.18 and above.
```
